tests/test_app.py

```diff
@@ -8,6 +9,20 @@ def test_init():
     assert app
 
 
+def test_run():
+    app = Application()
+    app.run_internally = mock.MagicMock(return_value=True)
+    assert app.run()
+    assert app.run_internally.called
+
+
+def test_run_with_exception():
```

Comment:
Mock is "magical". It's good idea to avoid it when possible and only use it if no better way exists.